habitat/datasets/object_nav/object_nav_dataset.py

```diff
@@ -63,6 +79,25 @@ def from_json(
             self.category_to_scene_annotation_category_id.keys()
         ), "category_to_task and category_to_mp3d must have the same keys"
 
+        if not "goals_by_category" in deserialized:
+            if len(deserialized["episodes"]) == 0:
+                return
+            else:
```

Comment:
I was expecting the code will create `goals_by_category` if it's not provided and make it new format.